File: latihan 12/Pengembalian.php
<?php
//Simpanlah dengan nama file : Pengembalian.php
require_once 'database.php';
class Pengembalian 
{
    private $db;
    private $table = 'pengembalian';
    public $id_anggota = "";
    public $id_buku = "";
    public $tanggal_pengembalian = "";
    public $denda = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_id_buku(int $id_buku)
    {
        $query = "SELECT * FROM $this->table WHERE id_buku = $id_buku";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`id_anggota`,`id_buku`,`tanggal_pengembalian`,`denda`) VALUES ('$this->id_anggota','$this->id_buku','$this->tanggal_pengembalian','$this->denda')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET id_anggota = '$this->id_anggota', id_buku = '$this->id_buku', tanggal_pengembalian = '$this->tanggal_pengembalian', denda = '$this->denda' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_id_buku($id_buku): int
    {
        $query = "UPDATE $this->table SET id_anggota = '$this->id_anggota', id_buku = '$this->id_buku', tanggal_pengembalian = '$this->tanggal_pengembalian', denda = '$this->denda' 
        WHERE id_buku = $id_buku";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_id_buku($id_buku): int
    {
        $query = "DELETE FROM $this->table WHERE id_buku = $id_buku";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: latihan 12/Pengembalian.py
import requests
import json
class Pengembalian:
    def __init__(self):
        self.__id=None
        self.__id_anggota = None
        self.__id_buku = None
        self.__tanggal_pengembalian = None
        self.__denda = None
        self.__url = "http://localhost/perpustakaan/pengembalian_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def id_anggota(self):
        return self.__id_anggota
        
    @id_anggota.setter
    def id_anggota(self, value):
        self.__id_anggota = value
    @property
    def id_buku(self):
        return self.__id_buku
        
    @id_buku.setter
    def id_buku(self, value):
        self.__id_buku = value
    @property
    def tanggal_pengembalian(self):
        return self.__tanggal_pengembalian
        
    @tanggal_pengembalian.setter
    def tanggal_pengembalian(self, value):
        self.__tanggal_pengembalian = value
    @property
    def denda(self):
        return self.__denda
        
    @denda.setter
    def denda(self, value):
        self.__denda = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_id_buku(self, id_buku):
        url = self.__url+"?id_buku="+id_buku
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__id_anggota = item['id_anggota']
            self.__id_buku = item['id_buku']
            self.__tanggal_pengembalian = item['tanggal_pengembalian']
            self.__denda = item['denda']
        return data
    def simpan(self):
        payload = {
            "id_anggota":self.__id_anggota,
            "id_buku":self.__id_buku,
            "tanggal_pengembalian":self.__tanggal_pengembalian,
            "denda":self.__denda
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_id_buku(self, id_buku):
        url = self.__url+"?id_buku="+id_buku
        payload = {
            "id_anggota":self.__id_anggota,
            "id_buku":self.__id_buku,
            "tanggal_pengembalian":self.__tanggal_pengembalian,
            "denda":self.__denda
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_id_buku(self,id_buku):
        url = self.__url+"?id_buku="+id_buku
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
